### PR TITLE
Add http loader fallback support 

### PR DESCRIPTION
**Pull request by victorjkhalaf**
Addresses issue https://github.com/thumbor/thumbor-aws/issues/44

I noticed the HTTP Loader support from tc-aws was missing and for those who may want to have a fallback loader when they may need to support http sources while supporting S3 this could be useful.

This ports more or less the same code from tc-aws and adapts it slightly for the updated format in this implementation. 

As I am relatively inexperienced with Python and don't have any testing experience in it, I am not sure if my adding of the tests from tc-aws will work and would appreciate any review/assistance in making them work!
I tried to run `make ci` after following the Contributing guidelines and too many tests were failing that were not mine. I am therefore unable to conclude if I had incorrectly run the testing or if this is the current state of the tests for this project.
I apologize in advance for that. 

I updated the toml file to add the mock library that was missing for the 2 tests to run. Again since this is a copy from tc-aws, it is possible I am not factoring in better ways to mock the thumbor http loader.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ pytest-xdist = "^2.5.0"
 yanc = "^0.3.3"
 pyssim = "^0.4.0"
 scipy = "^1.7.3"
+mock = "^4.0.3"
```
Comment (guilhermef) [inline pyproject.toml:52]: @victorjkhalaf `mock` is part of unittest: https://docs.python.org/3/library/unittest.mock.html#patch
There's no need to add it as a dependency.

**Review by guilhermef — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (victorjkhalaf) [reply pyproject.toml:52]: Strange, it had struggled to run at the time but I'll take it out!

Comment (victorjkhalaf) [inline pyproject.toml:52]:
```suggestion
```

Comment (guilhermef) [reply pyproject.toml]: It was added on python 3.3, make sure that you're using at least Python 3.7.

File: tests/test_loader.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 from preggy import expect
 from thumbor.config import Config
 from tornado.testing import gen_test
-
+from mock import patch
```
Comment (guilhermef) [inline tests/test_loader.py:17]: You need to replace this import with `from unittest.mock import patch`

Comment (victorjkhalaf) [reply tests/test_loader.py]: Done! thanks for the pointer

**Comment by victorjkhalaf** [comment]
@guilhermef excuse the delay. I have fixed the tests in [`d868211` (#45)](https://github.com/thumbor/thumbor-aws/pull/45/commits/d8682117223ab4b0afba531efa67c020a0ce2351) and it should pass now. Can you do a quick pass over the code? Again my Python experience is very limited so I apologize if it's not up to the standards. I welcome any feedback.

**Comment by guilhermef** [comment]
Hi @victorjkhalaf, the lint is failing now.
I've fixed the workflow trigger on main.

**Comment by victorjkhalaf** [comment]
@guilhermef I've fixed the lint issues reported. I'm not sure if it will run the build automatically this time, I took a look and it seems to still require you approve the workflow?

**Comment by guilhermef** [comment]
@victorjkhalaf I'm sorry, that's enforced for all first-time contributors on a project with GH actions.
I've approved the workflow run.

**Comment by victorjkhalaf** [comment]
@guilhermef my bad, it looks like there were two last linting errors. Should be fine now.

**Comment by victorjkhalaf** [comment]
@guilhermef I ended up running the linting step locally since it looks like I keep running into more linting issues 

It has passed for me so hopefully this is the last time I ask you to rerun for linting issues!

**Comment by guilhermef** [comment]
@victorjkhalaf You need to ensure that the mocked function is still an async function: https://dino.codes/posts/mocking-asynchronous-functions-python/

**Comment by victorjkhalaf** [comment]
@guilhermef I've attempted to mimic the same setup as the article using the 3.7- approach and it does not pass the mock_loader into the test. All 4 version builds are failing with the same issue: 
```
==============================================================================
ERROR: tests/test_loader.py::LoaderTestCase::test_should_use_http_loader
------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/thumbor-aws/thumbor-aws/.venv/lib/python3.10/site-packages/tornado/testing.py", line 98, in __call__
    result = self.orig_method(*args, **kwargs)
  File "/home/runner/work/thumbor-aws/thumbor-aws/.venv/lib/python3.10/site-packages/tornado/testing.py", line 599, in post_coroutine
    return self.io_loop.run_sync(
  File "/home/runner/work/thumbor-aws/thumbor-aws/.venv/lib/python3.10/site-packages/tornado/ioloop.py", line 530, in run_sync
    return future_cell[0].result()
  File "/home/runner/work/thumbor-aws/thumbor-aws/.venv/lib/python3.10/site-packages/tornado/ioloop.py", line 492, in run
    result = func()
  File "/home/runner/work/thumbor-aws/thumbor-aws/.venv/lib/python3.10/site-packages/tornado/testing.py", line 583, in pre_coroutine
    result = f(self, *args, **kwargs)
TypeError: LoaderTestCase.test_should_use_http_loader() missing 1 required positional argument: 'mock_loader'
```

The implementation of that article's code adapted to my test is on this line https://github.com/thumbor/thumbor-aws/pull/45/files#diff-bef3907b4316238adbc3b1efe0f2a9e572edd5a0e5e6d690d75890c0d4b6843eR68

I'm not sure why it isn't passing it other than a guess that it doesn't work when `self` is passed in first.

**Pull request closed without merging**